File: 2021/Day1.py
def part1(input):
    data = input.split("\n")
    increases = 0
    for i in range(1, len(data)):
        if int(data[i]) > int(data[i-1]):
            increases += 1
    return increases

def part2(input):
    data = input.split("\n")
    increases = 0
    for i in range(3, len(data)):
        if int(data[i]) > int(data[i-3]):
            increases += 1
    return increases

with open("input.txt") as f:
    input = f.read()
    print(f"Part 1 Answer: {part1(input)}")
    print(f"Part 2 Answer: {part2(input)}")

File: 2025/Day1.js
function part1(input) {
    const data = input.split("\n");
    let pos = 50;
    let zeroes = 0;
    for (let rot of data) {
        let dir = rot[0] == "R" ? 1 : -1;
        let val = parseInt(rot.slice(1));
        pos += dir * val;
        if (pos % 100 === 0) {
            zeroes++;
        }
    }
    return zeroes;
}

function part2(input) {
    const data = input.split("\n");
    let pos = 50;
    let zeroes = 0;
    for (let rot of data) {
        let dir = rot[0] == "R" ? 1 : -1;
        let val = parseInt(rot.slice(1));
        for (let i = 0; i < val; i++) {
            pos += dir;
            if (pos % 100 === 0) {
                zeroes++;
            }
        }
    }
    return zeroes;
}

const fs = require('fs');
fs.readFile('input.txt', 'utf8', (err, input) => {
    if (err) {
        console.error('Error reading input:', err);
        return;
    }
    console.log(`Part 1 Answer: ${part1(input)}`);
    console.log(`Part 2 Answer: ${part2(input)}`);
});